sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/StronglyTypedRecognizedForm.java

```diff
@@ -41,36 +41,40 @@ public static void main(final String[] args) {
             System.out.printf("----------- Recognized receipt info for page %d -----------%n", i);
             // Use Receipt model transform the recognized form to strongly typed US receipt fields
             Receipt usReceipt = new Receipt(recognizedForm);
-            System.out.printf("Merchant Name: %s, confidence: %.2f%n", usReceipt.getMerchantName().getValue(),
+            System.out.printf("Merchant Name: %s, confidence: %.2f%n", usReceipt.getMerchantName().getFieldValue(),
                 usReceipt.getMerchantName().getConfidence());
             System.out.printf("Merchant Address: %s, confidence: %.2f%n",
-                usReceipt.getMerchantAddress().getValue(),
+                usReceipt.getMerchantAddress().getFieldValue(),
                 usReceipt.getMerchantAddress().getConfidence());
+            // It will have to be this way?
+            // System.out.printf("Merchant Address: %s, confidence: %.2f%n",
+            //     usReceipt.getMerchantAddress().getFieldValue().asString(),
+            //     usReceipt.getMerchantAddress().getConfidence());
```

Comment:
@srnagar Confirming if this how we would be using the `FormField<T>` for strongly typed examples?

Reply:
I don't think the generic type `<T>` is useful if the user still has to do `getFieldValue().asString()`. I think we are trying to make `FormField` work both ways and is causing this issue.  

Have you considered the approach below? It's not ideal but a new type that's defined once can reduce the burden on the user. They'll not have to repeatedly call with `getValue().asType()` methods which is going to be accessed a lot more frequently.

```java

// Receipt type
public final class Receipt {
    private ReceiptType receiptType;
    private TypedFormField<String> merchantAddress; // return a different form field type
    
    public Receipt(RecognizedForm recognizedForm) {

       for (Map.Entry<String, FormField<?>> entry : recognizedForm.getFields().entrySet()) {
            String key = entry.getKey();
            FormField<?> fieldValue = entry.getValue();
            switch (key) {
                case "MerchantAddress":
                    merchantAddress = new TypedFormField(fieldValue, String.class);
                    break;
                
                default:
                    break;
            }
        }
        
    }

    // getter and setters here
    public TypedFormField<String> getMerchantAddress() {
         return merchantAddress;
    }
}

-------------------------------------------

// Strongly typed form field
public class TypedFormField<T> { // with a better class name

    private final FormField formField;
    private final Class<T> type;

    public TypedFormField(FormField formField, Class<T> type) {
        this.formField = formField;
        this.type = type;
    }

    public T getValue() {
        if (formField.getValueType() == FieldValueType.STRING && type.getClass().equals(String.class)) {
            return formField.getValue().asDate();
        }
        throw new IllegalStateException("Type mismatch error");
    }
   
    public float getConfidence() {
        return this.formField.getConfidence();
    }
 
}
-------------------------------------------
// User code
Receipt usReceipt = new Receipt(recognizedForm);
String address = usReceipt.getMerchantAddress().getValue()
float confidence = usReceipt.getMerchantAddress().getConfidence();
```